src/app/api/submissions/latest/route.ts

```diff
@@ -0,0 +1,90 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+// Helper to check if the origin is allowed
+function isAllowedOrigin(origin: string | null) {
+  if (!origin) return false;
+  try {
+    const { hostname } = new URL(origin);
+    return hostname.endsWith('.coderbase.dev') || hostname === 'coderbase.dev';
+  } catch {
+    return false;
+  }
+}
+
+// Placeholder for your actual data fetching logic
+async function getLatestSubmissionByEmail(email: string) {
+  // TODO: Replace with real data fetching (e.g., from Supabase)
+  return { id: 'example-id', email, content: 'Latest submission', createdAt: new Date().toISOString() };
+}
+
+function corsHeaders(origin: string | null): Record<string, string> {
```

Comment:
Should we move these helpers to a shared utility file (e.g., src/utils/cors.ts) so they can be reused across multiple API routes?